src/app/campaign/creative/creative-list.component.ts

```diff
@@ -0,0 +1,137 @@
+import { Component, OnInit, OnDestroy, ChangeDetectionStrategy } from '@angular/core';
+import { Router } from '@angular/router';
+import { MatSelectionListChange } from '@angular/material';
+import { Store, select } from '@ngrx/store';
+import { Observable, Subject, Subscription } from 'rxjs';
+import { first, withLatestFrom, debounceTime, distinctUntilChanged, filter } from 'rxjs/operators';
+import { Creative, CreativeParams } from '../store/models';
+import {
+  selectCreativeListPageOrderedByCreatedAt,
+  selectRoutedCampaignId,
+  selectRoutedFlightId,
+  selectRoutedFlightZoneId,
+  selectCreativeParams,
+  selectCreativeTotal
+} from '../store/selectors';
+import * as campaignActions from '../store/actions/campaign-action.creator';
+import * as creativeActions from '../store/actions/creative-action.creator';
+
+@Component({
+  template: `
+    <mat-toolbar color="primary">
+      <mat-toolbar-row>
+        <mat-icon>attach_file</mat-icon>
+        <h2>Add Existing Creative</h2>
+        <button mat-button aria-label="cancel" (click)="onCancel()"><mat-icon>clear</mat-icon></button>
+      </mat-toolbar-row>
+    </mat-toolbar>
+    <section *ngIf="creativeParams$ | async as params">
+      <h3>Filter Creatives</h3>
+      <mat-form-field appearance="outline">
+        <input matInput type="text" placeholder="Search by filename" (keyup)="onSearch($event.target.value)" />
+      </mat-form-field>
+      <h3>Add Creative</h3>
+      <mat-selection-list (selectionChange)="onSelectCreatives($event)" multiple="true">
+        <mat-list-option
+          *ngFor="let creative of creativeList$ | async"
+          checkboxPosition="before"
+          [value]="creative.id"
+          [title]="creative.filename"
+          [selected]="selectedCreativeIds[params.page] && selectedCreativeIds[params.page].indexOf(creative.id) > -1"
+        >
+          <h4 matLine>{{ creative.filename }}</h4>
+          <p matLine>
+            <span class="advertiser">{{ creative.advertiser?.name }}</span>
+            <span class="uploaded">Uploaded {{ creative.createdAt | date: 'MM/dd/yyyy':'UTC' }}</span>
+          </p>
+        </mat-list-option>
+      </mat-selection-list>
+      <mat-paginator
+        [length]="creativeTotal$ | async"
+        [pageIndex]="params.page - 1"
+        [pageSize]="params.per"
+        [pageSizeOptions]="[5, 10, 20]"
+        (page)="onPage({ page: $event.pageIndex + 1, per: $event.pageSize })"
+      ></mat-paginator>
+      <div class="form-submit">
```

Comment:
Update: Realized I forgot to add the Save/Cancel buttons as shown in the design. https://github.com/PRX/grove.prx.org/issues/236 When I've shown this in demos, I had it so you just clicked to select a single creative and it would be added to the zone and the drawer closed without any additional buttons. The MatSelectionList supports single selection, but it still looks like a checkbox rather than a radio button as we have in the design. Having the save button down below also made me think choosing multiple creatives and clicking a button would be more appropriate, and of course that had to work across this paged list.